core/audits/prioritize-lcp-image.js

```diff
@@ -131,7 +133,31 @@ class PrioritizeLcpImage extends Audit {
     // Get last candidate, in case there was more than one.
     }).sort((a, b) => b.ts - a.ts)[0];
 
-    return lcpImagePaintEvent?.args.data?.imageUrl;
+    const lcpUrl = lcpImagePaintEvent?.args.data?.imageUrl;
+    if (!lcpUrl) return;
+
+    const candidates = networkRecords.filter(record => {
+      return record.url === lcpUrl &&
+          record.finished &&
+          // Same frame as LCP trace event.
+          record.frameId === lcpImagePaintEvent.args.frame &&
+          record.networkRequestTime < (processedNavigation.timestamps.largestContentfulPaint || 0);
+    }).map(record => {
+      // Follow any redirects to find the real image request.
+      while (record.redirectDestination) {
+        record = record.redirectDestination;
+      }
+      return record;
+    }).filter(record => {
+      // Don't select if also loaded by some other means (xhr, etc). `resourceType`
+      // isn't set on redirect _sources_, so have to check after following redirects.
+      return record.resourceType === 'Image';
+    });
+
+    // If there are still multiple candidates, at this point it appears the page
+    // simply made multiple requests for the image. The first loaded is the best
+    // guess of the request that made the image available for use.
```

Comment:
this happens more often than you'd think it would, so there's not much that can be done besides pick one. Maybe we'll find more criteria in the future that will further reduce real duplicate candidate cases